### DISCUSSION TITLE
Specific PC and browser issue

### DISCUSSION THREAD

**Discussion by PeopleInside** [LiveHelperChat/livehelperchat#2139] [General]
Hi,I hope the day is going well.
I write here for asking a suggestion for understand an issue that seems affect only my PC.

If I visit my website https://www.peopleinside.it from my Windows 11 Pro Edge browser version 128.0.2739.67 64 bit the widget miss the HTML part.

I'm unable to replicate this on Edge run on Windows sandbox or on smartphone.
I deleted cookie, cache, I reset and unistalled Edge, reinstalled but the issue persist.
I also tried on a different Edge profile, different Windows PC. Issue still persist.

From sandbox, Internet if I test the chat link in a edge browser, from smartphone no issues.

This is how the widget appear on my PC Edge:
![immagine](https://github.com/user-attachments/assets/cd45e493-d51c-40c1-bba9-ecbf03464525)

This is how should be and it is on Windows sandbox Edge and on Firefox:
![immagine](https://github.com/user-attachments/assets/a74aa4db-4e0b-4137-be85-8cc14c75735e)

How can I understand why the HTML part is missing?
I tried to check browser console when loading the widget address but I don't see anything.. I'm not an expert of programming or of the developer browser console.

Any idea on how I can understand why on Edge of my PC the HTML pre-chat is not showed?

![00001](https://github.com/user-attachments/assets/8ccbf0d0-d115-4e77-99dc-db9a639039ee)

**Comment by remdex** [comment]
Sorry, but your configuration - your issues... Especially you can't wait explain properly.

**Reply by PeopleInside** [reply]
Ok no problem, I don't wanna stress you as I also know and think you can be busy.
Mine was just a way to ask and discuss.

Maybe from your reply I can think my message was not clear.

Simply on my Windows PC, on the Edge browser only, 
when I open, as visitor, my widget chat, 
the HTML that I have before the input field is not showed.

Above i posted a screenshot with a short widget from my Edge that is not showing the online and offline HTML and after the image from Firefox where all works.

This issue I think seems not related to live helper chat as if I test my chat widget in Edge on the smartphone or opening a Windows virtual machine (sandbox) the widget is showed correctly with the full HTML text introduction as in the second screenshot.

Reset cookies, Edge, clear all history, unistall Edge and reinstall never help.
I'm try now to asking on webcompat or to people that can maybe give a suggestion what can be. I tried also on incognito and without any browser extension.

This issue is just in my browser so seems not a live helper chat related issue but the strange thing is that is happening with the live helper chat widget and I cannot understand why just the HTML before the input field are not showed :-)

The good news is for now seems my PC is the only that show this issue on the Edge browser.
If I open a virtual machine on my PC this issue is not present so maybe an issue with Edge that is not working well. Don't know.

If I will figure out I will try to update the discussion :)

**Comment by remdex** [comment]
Also you should try to clear your cookies on browser it does not work.

**Reply by PeopleInside** [reply]
Done but did not help. Seems an issue with my Edge browser.
I will try to figure out :-)

**Comment by PeopleInside** [comment]
I was able to replicate the issue also on Firefox and also the issue moved from a domain to another one in Firefox.

You have implemented recently language translator for HTML messages. I suppose the issue is created by that.
Seems the detection not always work, this is why since long time i don't use embed code detect visitor language.

If the browser language is well detected the HTML is showed, if something get wrong in the detection then HTML is not showed.

I see the issue present on my peopleinside.it domain in Edge, now I was able to see the issue for marcoborla.it also in Firefox... than this helped me to understand that issue are caused by the auto detection and the fact HTML must understand who language speak the visitor for show the HTML.

There is no way to turn off the translation and show an HTML in the widget for all languages?

**Comment by PeopleInside** [comment]
Finally I discovered the issue is caused by the detection of language that goes in tilt on Edge and on Firefox too after a while.
I don't know what cause this but seems if a unknow language is detected no HTML is showed.
Should be present an HTML option that is showed for not detected language. There is the main tab but seems this is not used if the detected language is unknow as at the start you introduced this new translation I need to use the translation tab to add Italian and English.

There is a way to turn off this translation detection and always show the HTML to all?
This is a good topic for a new issue as now my widget HTML with this translation system never work always.

https://app.screencast.com/zOWxm6f7YZ051

I tried to remove translations and leave only main to show always to all languages my HTML but is not working
https://app.screencast.com/Syjaz50KCq8Ff

And if I use translations it works but not always as showed in the first video.

--
If you don't want fix this could you please indicate the commit where introduced the translation in the HTML so I can try to return back to the old system with override template? But maybe this will turn some old bugs on like the HTML not showed on widget status.. or maybe not I don't remember if you fixed all and then after introduced this translation.

A fix can be show the HTML tab in the main to all undetected languages and if no translation is used (selected). If HTML in main tab and no language option selected for translations then HTML is showed always for all languages.

**Reply by remdex** [reply]
Thats not an issue. You should in your widget theme just put html in languages you want it to be shown 😀 

https://doc.livehelperchat.com/docs/theme#multilanguage-flow

or just enter siteaccess you want it to be valid here 


https://github.com/LiveHelperChat/livehelperchat/blob/7b62d06fdbc2e4048672c4567a0d09aeb8cfc56b/lhc_web/settings/settings.ini.default.php#L29

**Reply by PeopleInside** [reply]
In my [settings.ini.default.php](https://github.com/LiveHelperChat/livehelperchat/blob/7b62d06fdbc2e4048672c4567a0d09aeb8cfc56b/lhc_web/settings/settings.ini.default.php#L29) on line 24 I have `'default_site_access_list' => ['eng'],`
Umh my default language should be Italian but what I have to put here?

Also if I have this line in the settings, when I go on settings, live helper configuration, widget themes, custom content I still have to add translation for all languages. If I insert the HTML only on main tab... the HTML is not showed
![immagine](https://github.com/user-attachments/assets/94e13209-d893-487d-9537-4c57845cc90b)

Suppose you tell me the right default language to set in settings.ini.default.php like
`'default_site_access_list' => ['ita'],`

['ita'] is correct for Italian?

This also will never solve the issue I'm reporting because I already added HTML for all languages.
I discovered in Edge there is two Italian language, one works and other one will be never detected or know from LHC so no HTML will be showed. The issue happened also on Firefox.

So the issue is that seems there is no way to setup LHC to show an HTML for undetected language.

My chat widget must support only two language: Italian and English.
Now suppose I insert the translation in the theme widget content for Italian and English but a Cinese visitor open my website.
LHC will never show the HTML because the visitor browser is not Italian and is not English.

What would be the solution?
For avoid this I inserted a translation for all available languages but still having the issue if you saw my two videos above.

![immagine](https://github.com/user-attachments/assets/674c91c3-d19e-46ff-868c-7fd03723b335)

Instead of never show any HTML, the main tab HTML should be showed when language is not present in LHC.

What is the commit to disable this translation feature that has been added in the HTML widget field?
If a fix cannot fix this issue I need find a way to disable this language detection and show always the same HTML. This translation feature create issues to me as is not working well on my test.

**Comment by PeopleInside** [comment]
OK maybe resolved by set `'default_site_access_list' => ['ita'],`
I never had this issue in the past, some update create a new issues with HTML and translation. I hope now will work..

But if the live chat widget is inserted in a Italian and in a English website?
If the default language is set to Italian means the widget will not work on English websites?

I proffered when this kind of language detection was not present in the HTML widget ^_^ or just want turn it off.

**Reply by remdex** [reply]
You can put both ita,eng...

**Reply by PeopleInside** [reply]
umh thanks.
For now all seems work.
Hoe to put both?

Like: 
'default_site_access_list' => ['ita eng'],

or

'default_site_access_list' => ['ita, eng'], ?

Many thanks.

**Reply by remdex** [reply]
Educate yourself :) ['ita','eng']

**Reply by PeopleInside** [reply]
> Educate yourself :) ['ita','eng']

I search in the documentation but I found this not explained.
Thanks
